h2o-k8s/src/main/java/water/k8s/lookup/KubernetesDnsLookup.java

```diff
@@ -0,0 +1,159 @@
+package water.k8s.lookup;
+
+
+import water.util.Log;
+
+import javax.naming.Context;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.naming.directory.Attribute;
+import javax.naming.directory.Attributes;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.InitialDirContext;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.*;
+import java.util.regex.Pattern;
+
+/**
+ * Discovery strategy on a DNS cluster leveraging the DNS record of a Kubernetes headless service present in the cluster.
+ * Kubernetes headless services, instead of load-balancing the requests onto one of the underlying pods, return the
+ * addresses of all the pods covered by the headless service.
+ * <p>
+ * Such pods can then be discovered via the above-mentioned DNS record. In order for H2O to know which service to query,
+ * it is mandatory for the K8S user to pass the name of the headless service to the H2O container, as follows:
+ *
+ * <pre>
+ * apiVersion: apps/v1
+ * kind: StatefulSet
+ * metadata:
+ *   name: example
+ *   namespace: h2o-statefulset
+ * spec:
+ *   serviceName: h2o-service
+ *   replicas: 3
+ *   selector:
+ *     matchLabels:
+ *       app: k8s-env
+ *   template:
+ *     metadata:
+ *       labels:
+ *         app: k8s-env
+ *     spec:
+ *       terminationGracePeriodSeconds: 10
+ *       containers:
+ *         - name: k8s-env
+ *           image: 'pscheidl/h2o-k8s'
+ *           ports:
+ *             - containerPort: 54321
+ *               protocol: TCP
+ *             - containerPort: 54322
+ *               protocol: TCP
+ *           env:
+ *           - name: H2O_KUBERNETES_SERVICE_DNS
+ *             value: h2o-service.h2o-statefulset.svc.cluster.local
+ * </pre>
+ */
+public class KubernetesDnsLookup implements KubernetesLookup {
+
+    private static final String H2O_K8S_SERVICE_DNS_KEY = "H2O_KUBERNETES_SERVICE_DNS";
```

Comment:
nit: this is a bit confusing... at least to me...

Reply:
If it's confusing, then it's not nitpicking. I'll fix that !

Reply:
well it made sense the second time I looked at it. So I figured it was a little nitpicky.